### PR TITLE
Animated smokes

### PR DESCRIPTION
**Pull request by Entity-y**
#### Summary
Ultica "animated smokes"

#### Content of the change
Animated smoke sprites for wielded cigarettes, cigars, joints. Also adds wielded roaches and butts.

#### Testing

--Smoking sprites are stuck on the default lit sprite and are not making use of the animated sprites--

Other sprites are working fine, tested on male and female.

Joint
imageimageimage
Cig
imageimageimage
Cigar - created by MrAsbestos, I only added smoke and cigar butt
imageimageimage
Concept of smoke animation when animation issue is resolved - cigarettes have been adjusted, see screenshots
smoking
Untitled 09-07-2021 04-25-47 (2)
Untitled 09-07-2021 04-25-47 (1)

### CODE REVIEW

**Comment by Entity-y** [comment]
not sure what to change in the json file to make animations work properly

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigar/cigar.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+[
+  {
+    "id": "overlay_female_wielded_cigar_lit",
+    "fg": "overlay_female_wielded_cigar_lit",
+    "multitile": true,
+    "additional_tiles": [
+      {
+        "id": "overlay_female_wielded_cigar_lit",
+        "animated": true,
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigar/cigar.json:9]:
```suggestion
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigar/cigar.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
+[
+  {
+    "id": "overlay_female_wielded_cigar_lit",
+    "fg": "overlay_female_wielded_cigar_lit",
+    "multitile": true,
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigar/cigar.json:5]:
```suggestion
    "animated": true,
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigar/cigar.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    ]
+  },
+  {
+    "id": "overlay_male_wielded_cigar_lit",
+    "fg": "overlay_male_wielded_cigar_lit",
+    "multitile": true,
+    "additional_tiles": [
+      {
+        "id": "overlay_male_wielded_cigar_lit",
+        "animated": true,
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigar/cigar.json:29]:
```suggestion
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigar/cigar.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+          { "weight": 14, "sprite": "overlay_female_wielded_cigar_lit_6" },
+          { "weight": 14, "sprite": "overlay_female_wielded_cigar_lit_7" }
+        ]
+      }
+    ]
+  },
+  {
+    "id": "overlay_male_wielded_cigar_lit",
+    "fg": "overlay_male_wielded_cigar_lit",
+    "multitile": true,
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigar/cigar.json:25]:
```suggestion
    "animated": true,
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigarette/cigarette.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
+[
+  {
+    "id": "overlay_female_wielded_cig_lit",
+    "fg": "overlay_female_wielded_cig_lit",
+    "multitile": true,
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigarette/cigarette.json:5]:
```suggestion
    "animated": true,
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigarette/cigarette.json [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    "additional_tiles": [
+      {
+        "id": "overlay_female_wielded_cig_lit",
+        "animated": true,
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigarette/cigarette.json:9]:
```suggestion
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigarette/cigarette.json [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+          { "weight": 14, "sprite": "overlay_female_wielded_cig_lit_6" },
+          { "weight": 14, "sprite": "overlay_female_wielded_cig_lit_7" }
+        ]
+      }
+    ]
+  },
+  {
+    "id": "overlay_male_wielded_cig_lit",
+    "fg": "overlay_male_wielded_cig_lit",
+    "multitile": true,
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigarette/cigarette.json:25]:
```suggestion
    "animated": true,
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigarette/cigarette.json [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    "additional_tiles": [
+      {
+        "id": "overlay_male_wielded_cig_lit",
+        "animated": true,
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigarette/cigarette.json:29]:
```suggestion
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigar/cigar.json [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    "additional_tiles": [
+      {
+        "id": "overlay_male_wielded_cigar_lit",
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigar/cigar.json:28]:
```suggestion
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigar/cigar.json [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+          { "weight": 14, "sprite": "overlay_male_wielded_cigar_lit_1" },
+          { "weight": 14, "sprite": "overlay_male_wielded_cigar_lit_2" },
+          { "weight": 14, "sprite": "overlay_male_wielded_cigar_lit_3" },
+          { "weight": 14, "sprite": "overlay_male_wielded_cigar_lit_4" },
+          { "weight": 14, "sprite": "overlay_male_wielded_cigar_lit_5" },
+          { "weight": 14, "sprite": "overlay_male_wielded_cigar_lit_6" },
+          { "weight": 14, "sprite": "overlay_male_wielded_cigar_lit_7" }
+        ]
+      }
+    ]
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigar/cigar.json:40]:
```suggestion
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigar/cigar.json [hunk 11]
```diff
@@ -0,0 +1,42 @@
+[
+  {
+    "id": "overlay_female_wielded_cigar_lit",
+    "fg": "overlay_female_wielded_cigar_lit",
+    "multitile": true,
+    "additional_tiles": [
+      {
+        "id": "overlay_female_wielded_cigar_lit",
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigar/cigar.json:8]:
```suggestion
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigar/cigar.json [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+          { "weight": 14, "sprite": "overlay_female_wielded_cigar_lit_1" },
+          { "weight": 14, "sprite": "overlay_female_wielded_cigar_lit_2" },
+          { "weight": 14, "sprite": "overlay_female_wielded_cigar_lit_3" },
+          { "weight": 14, "sprite": "overlay_female_wielded_cigar_lit_4" },
+          { "weight": 14, "sprite": "overlay_female_wielded_cigar_lit_5" },
+          { "weight": 14, "sprite": "overlay_female_wielded_cigar_lit_6" },
+          { "weight": 14, "sprite": "overlay_female_wielded_cigar_lit_7" }
+        ]
+      }
+    ]
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigar/cigar.json:20]:
```suggestion
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigarette/cigarette.json [hunk 13]
```diff
@@ -0,0 +1,42 @@
+[
+  {
+    "id": "overlay_female_wielded_cig_lit",
+    "fg": "overlay_female_wielded_cig_lit",
+    "multitile": true,
+    "additional_tiles": [
+      {
+        "id": "overlay_female_wielded_cig_lit",
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigarette/cigarette.json:8]:
```suggestion
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigarette/cigarette.json [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+          { "weight": 14, "sprite": "overlay_female_wielded_cig_lit_1" },
+          { "weight": 14, "sprite": "overlay_female_wielded_cig_lit_2" },
+          { "weight": 14, "sprite": "overlay_female_wielded_cig_lit_3" },
+          { "weight": 14, "sprite": "overlay_female_wielded_cig_lit_4" },
+          { "weight": 14, "sprite": "overlay_female_wielded_cig_lit_5" },
+          { "weight": 14, "sprite": "overlay_female_wielded_cig_lit_6" },
+          { "weight": 14, "sprite": "overlay_female_wielded_cig_lit_7" }
+        ]
+      }
+    ]
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigarette/cigarette.json:20]:
```suggestion
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigarette/cigarette.json [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+  },
+  {
+    "id": "overlay_male_wielded_cig_lit",
+    "fg": "overlay_male_wielded_cig_lit",
+    "multitile": true,
+    "additional_tiles": [
+      {
+        "id": "overlay_male_wielded_cig_lit",
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigarette/cigarette.json:28]:
```suggestion
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigarette/cigarette.json [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+          { "weight": 14, "sprite": "overlay_male_wielded_cig_lit_1" },
+          { "weight": 14, "sprite": "overlay_male_wielded_cig_lit_2" },
+          { "weight": 14, "sprite": "overlay_male_wielded_cig_lit_3" },
+          { "weight": 14, "sprite": "overlay_male_wielded_cig_lit_4" },
+          { "weight": 14, "sprite": "overlay_male_wielded_cig_lit_5" },
+          { "weight": 14, "sprite": "overlay_male_wielded_cig_lit_6" },
+          { "weight": 14, "sprite": "overlay_male_wielded_cig_lit_7" }
+        ]
+      }
+    ]
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/cigarette/cigarette.json:40]:
```suggestion
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/joint/joint.json [hunk 17]
```diff
@@ -0,0 +1,42 @@
+[
+  {
+    "id": "overlay_female_wielded_joint_lit",
+    "fg": "overlay_female_wielded_joint_lit",
+    "multitile": true,
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/joint/joint.json:5]:
```suggestion
    "animated": true,
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/joint/joint.json [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+    "additional_tiles": [
+      {
+        "id": "overlay_female_wielded_joint_lit",
+        "animated": true,
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/joint/joint.json:9]:
```suggestion
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/joint/joint.json [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+          { "weight": 14, "sprite": "overlay_female_wielded_joint_lit_1" },
+          { "weight": 14, "sprite": "overlay_female_wielded_joint_lit_2" },
+          { "weight": 14, "sprite": "overlay_female_wielded_joint_lit_3" },
+          { "weight": 14, "sprite": "overlay_female_wielded_joint_lit_4" },
+          { "weight": 14, "sprite": "overlay_female_wielded_joint_lit_5" },
+          { "weight": 14, "sprite": "overlay_female_wielded_joint_lit_6" },
+          { "weight": 14, "sprite": "overlay_female_wielded_joint_lit_7" }
+        ]
+      }
+    ]
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/joint/joint.json:20]:
```suggestion
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/joint/joint.json [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+  },
+  {
+    "id": "overlay_male_wielded_joint_lit",
+    "fg": "overlay_male_wielded_joint_lit",
+    "multitile": true,
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/joint/joint.json:25]:
```suggestion
    "animated": true,
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/joint/joint.json [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+    "additional_tiles": [
+      {
+        "id": "overlay_male_wielded_joint_lit",
+        "animated": true,
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/joint/joint.json:29]:
```suggestion
```

File: gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/joint/joint.json [hunk 22]
```diff
@@ -0,0 +1,42 @@
...
+          { "weight": 14, "sprite": "overlay_male_wielded_joint_lit_1" },
+          { "weight": 14, "sprite": "overlay_male_wielded_joint_lit_2" },
+          { "weight": 14, "sprite": "overlay_male_wielded_joint_lit_3" },
+          { "weight": 14, "sprite": "overlay_male_wielded_joint_lit_4" },
+          { "weight": 14, "sprite": "overlay_male_wielded_joint_lit_5" },
+          { "weight": 14, "sprite": "overlay_male_wielded_joint_lit_6" },
+          { "weight": 14, "sprite": "overlay_male_wielded_joint_lit_7" }
+        ]
+      }
+    ]
```
Comment (Fris0uman) [inline gfx/UltimateCataclysm/pngs_human_body_32x36/overlay/wielded/smokes/joint/joint.json:40]:
```suggestion
```

**Review by Fris0uman — COMMENTED** [review]
Here's how to do it, I might hav messed up som {} and [] because the github interface is not great to do it. But basically you don't need multitile, you don't need additional_tile, you just need animated: true

**Comment by acepleiades** [comment]
Well let's see what happens

**Comment by Fris0uman** [comment]
https://user-images.githubusercontent.com/41293484/132373878-b8120024-d4fe-45b9-a6d5-157078a5f52d.mp4


https://user-images.githubusercontent.com/41293484/132373886-84edb051-1753-402a-818d-cb26d7e2e8f9.mp4

it works

**Comment by acepleiades** [comment]
That is really excellent

**Pull request merged**